### PR TITLE
Neden sinir feature/messaging UI

### PR DESCRIPTION
**Pull request by NedenSinir**
# What I did
I implemented a complete messaging UI feature for the SkillBridge application. This includes:
- A new `MessageScreen` composable that displays messages in a conversation
- A `MessageViewModel` that manages the UI state and message operations
- Message input functionality with send capability
- Comprehensive UI and unit tests for the messaging feature

# How I did it
- **Created MessageScreen.kt**: Built a composable UI with:
  - A lazy column to display messages in reverse chronological order (latest at bottom)
  - Message bubbles that visually differentiate between current user and other user messages
  - A message input field with rounded corners supporting multiline text (up to 4 lines)
  - A send button that's enabled only when the message is not empty
  - Error handling and loading state displays

- **Created MessageViewModel.kt**: Implemented a ViewModel that:
  - Manages message loading from the repository
  - Handles message composition and sending
  - Maintains UI state including messages, current input, loading state, and errors
  - Integrates with `UserSessionManager` for authentication context
  - Automatically refreshes messages after sending

- **Added comprehensive tests**:
  - `MessageScreenTest.kt`: UI tests covering message display, input interactions, send functionality, empty states, and error handling
  - `MessageViewModelTest.kt`: Unit tests for ViewModel logic including message loading, sending, error handling, and edge cases

- **Removed old test file**: Deleted the deprecated `EndToEndM2.kt` test file (304 lines removed)

# How to verify it
1. Navigate to the messaging section of the app
2. Open a conversation with another user
3. Verify existing messages are displayed correctly with proper styling (different bubbles for sender vs receiver)
4. Type a message in the input field
5. Confirm the send button is disabled when empty and enabled when text is present
6. Tap send and verify the message appears in the conversation
7. Test multiline message support by typing multiple lines
8. Verify error messages display correctly if message loading or sending fails
9. Run the test suites to confirm all functionality works as expected

# Demo photos
image
image
image

# Pre-merge checklist
The changes I introduced:
- [x] work correctly
- [x] do not break other functionalities
- [x] work correctly on Android
- [x] are fully tested (or have tests added)

### CODE REVIEW

File: app/src/main/java/com/android/sample/ui/communication/MessageViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+  /** Updates the text for the new message being composed. */
+  fun onMessageChange(newMessage: String) {
+    _uiState.update { it.copy(currentMessage = newMessage) }
+  }
+
+  /** Sends the current message. */
+  fun sendMessage() {
+    val content = _uiState.value.currentMessage.trim()
+    if (content.isEmpty()) return
```
Comment (nahuelArthur) [inline app/src/main/java/com/android/sample/ui/communication/MessageViewModel.kt]: Think it is not necessary to check if it's empty, because you can only send not blank message (in the screen)

**Review by nahuelArthur — APPROVED** [review]
lgtm
